### PR TITLE
Encapsulate auth data extras

### PR DESCRIPTION
**Pull request by dimas-b**

### CODE REVIEW

File: persistence-api/src/main/java/io/stargate/db/AuthenticatedUser.java [hunk 1]
```diff
@@ -29,23 +36,92 @@
...
+      if (token == null || roleName == null) {
+        throw new IllegalStateException("token and roleName must be provided");
+      }
+
+      Builder<String, String> map = ImmutableMap.builder();
+      for (Entry<String, ByteBuffer> e : customPayload.entrySet()) {
+        String key = e.getKey();
+        if (key.startsWith(CUSTOM_PAYLOAD_NAME_PREFIX)) {
+          String name = key.substring(CUSTOM_PAYLOAD_NAME_PREFIX.length());
+          String value = StandardCharsets.UTF_8.decode(token).toString();
```
Comment (dougwettlaufer) [inline persistence-api/src/main/java/io/stargate/db/AuthenticatedUser.java]: Shouldn't this be either token or role?

**Review by dougwettlaufer — COMMENTED** [review]
Looks like the right direction

File: persistence-api/src/main/java/io/stargate/db/AuthenticatedUser.java (hunk 1 above)
Comment (dimas-b) [reply persistence-api/src/main/java/io/stargate/db/AuthenticatedUser.java]: fixed

File: restapi/src/main/java/io/stargate/web/resources/RowResource.java [hunk 2]
```diff
@@ -546,7 +558,8 @@ public Response updateRow(
       final RowUpdate changeSet) {
     return RequestHandler.handle(
         () -> {
-          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token);
+          Map<String, String> allHeaders = getAllHeaders(request);
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, allHeaders);
```
Comment (tomekl007) [inline restapi/src/main/java/io/stargate/web/resources/RowResource.java:562]: This is problematic. It calls the `authenticationService.validateToken(token, headers);` that in turn call the:
```
default AuthenticationSubject validateToken(String token, Map<String, String> headers)
      throws UnauthorizedException {
    return validateToken(token);
  }
```
and headers are ignored, and not propagated. Similar for other calls

**Review by tomekl007 — APPROVED** [review]

File: restapi/src/main/java/io/stargate/web/resources/RowResource.java (hunk 2 above)
Comment (dimas-b) [reply restapi/src/main/java/io/stargate/web/resources/RowResource.java:562]: Authentication implementations that need to examine headers are expected to override `validateToken(String token, Map<String, String> headers)`

Comment (dimas-b) [reply restapi/src/main/java/io/stargate/web/resources/RowResource.java:562]: the `default` method is a convenience for "simple" authenticators

**Review by dougwettlaufer — APPROVED** [review]

**Review by mpenick — APPROVED** [review]

**Pull request merged**